R/mfx-tidiers.R

```diff
@@ -0,0 +1,241 @@
+#' @templateVar class mfx
+#' @template title_desc_tidy
+#' 
+#' @description The particular functions below provide generic tidy methods for 
+#'   objects returned by the `mfx` package, preserving the calculated marginal 
+#'   effects instead of the naive model coefficients. The returned tidy tibble 
+#'   will also include an additional "atmean" column indicating how the marginal 
+#'   effects were originally calculated (see Details below).
+#' 
+#' @param x A `betamfx`, `logitmfx`, `negbinmfx`, `poissonmfx`, or `probitmfx`  object.
+#' @template param_confint
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(
+#'   "term",
+#'   atmean = "TRUE if the marginal effects were originally calculated as the partial effects for the average observation. If FALSE, then these were instead calculated as average partial effects.",
+#'   "estimate",
+#'   "std.error",
+#'   "statistic",
+#'   "p.value",
+#'   "conf.low",
+#'   "conf.high"
+#' )
+#' 
+#' @details The `mfx` package provides methods for calculating marginal effects
+#'   for various generalised linear models (GLMs). Unlike standard linear models 
+#'   (e.g. OLS regression), estimated model coefficients in a GLM cannot be
+#'   directly interpreted as marginal effects (i.e., the change in the response 
+#'   variable predicted after a one unit change in one of the regressors). This 
+#'   is because the estimated coefficients are multiplicative, dependent on both
+#'   the link function that was used for the estimation and any other variables
+#'   that were included in the model. When calculating marginal effects, users
+#'   must typically choose whether they want to use i) the average observation 
+#'   in the data, or ii) the average of the sample marginal effects. See 
+#'   `vignette("mfxarticle")` from the `mfx` package for more details.
+#'   
+#' @examples
+#' \dontrun{ 
+#' library(mfx)
+#' 
+#' ## Get the marginal effects from a logit regression
```

Comment:
Can you use `TRUE` instead of `T` throughout here? It's a good habit to get into.